File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/Activities/ElginPay/FormaFinanciamento.java
package com.elgin.java_intentdigitalhub_smartpos.Activities.ElginPay;

public enum FormaFinanciamento {
    FINANCIAMENTO_A_VISTA(1),
    FINANCIAMENTO_PARCELADO_EMISSOR(2),
    FINANCIAMENTO_PARCELADO_ESTABELECIMENTO(3);

    private final int codigoFormaParcelamento;

    FormaFinanciamento(int codigoFormaParcelamento) {
        this.codigoFormaParcelamento = codigoFormaParcelamento;
    }

    public int getCodigoFormaParcelamento() {
        return codigoFormaParcelamento;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/Activities/ElginPay/FormaFinanciamento.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.Activities.ElginPay

/**
 * Formas de financiamento e o valor a ser enviado em cada opção para os comandos
 */
enum class FormaFinanciamento(val codigoFormaParcelamento: Int) {
    FINANCIAMENTO_A_VISTA(1), FINANCIAMENTO_PARCELADO_EMISSOR(2), FINANCIAMENTO_PARCELADO_ESTABELECIMENTO(
        3
    );
}